src/heap_ptmalloc_2_35.cpp

```diff
@@ -972,8 +972,13 @@ thread_tcache (struct thread_info *info, void *data)
 		CA_PRINT("Assumed tcache size=%ld while gdb sees size=%ld\n", sizeof(tcps), valsz);
 		return false;
 	}
-	addr = value_as_address(val);
-	//CA_PRINT("tcache for ptid.pid [%d]: 0x%lx\n", info->ptid.pid(), addr);
+	// temp fix
+	tcache_perthread_struct *ptcps;
+
+	gdb::array_view<gdb_byte> content = value_contents_raw(val);
+	memcpy(&ptcps, content.data(), sizeof(ptcps));
+	//addr = value_as_address(val);
+	addr = (address_t)ptcps;	//CA_PRINT("tcache for ptid.pid [%d]: 0x%lx\n", info->ptid.pid(), addr);
```

Comment:
How about move this code into the try/catch clause as well?

Reply:
Yes. The fix would be added a catch for the `variable_as_address``